### PR TITLE
Rename nonevict to noevict branch 0.7

### PR DESCRIPTION
**Pull request by offthewall123**
## What changes were proposed in this pull request?

Rename nonevict to noevict branch 0.7

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/MemoryManager.scala [hunk 1]
```diff
@@ -104,10 +104,10 @@ private[sql] object MemoryManager extends Logging {
           logWarning(s"current spark.sql.oap.fiberCache.memory.manager: ${memoryManagerOpt} " +
             "takes no effect, use 'tmp' as memory manager for vmem cache instead.")
         }
-      case "nonevict" =>
+      case "noevict" =>
```
Comment (winningsix) [inline src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/MemoryManager.scala:107]: Could we define those cache strategies like a Emun? So we we can maintain the code more easily?

Comment (offthewall123) [reply src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/MemoryManager.scala:107]: Tried this.Seems not work in this scenario,cacheStrategyOpt is String and Scala only allows the same type in match...case...statements.

**Comment by winningsix** [comment]
Merged. Which JIRA ID should we close?

**Comment by offthewall123** [comment]
> Merged. Which JIRA ID should we close?

https://jira.devtools.intel.com/browse/POAE7-259
https://jira.devtools.intel.com/browse/POAE7-258

**Pull request merged**